jwst/tests_nightly/general/pipelines/test_nrc_tso3_1.py

```diff
@@ -0,0 +1,195 @@
+    step.outlier_detection.nhigh = 0
+    step.outlier_detection.maskpt = 0.7
+    step.outlier_detection.grow = 1
+    step.outlier_detection.snr = '4.0 3.0'
+    step.outlier_detection.scale = '0.5 0.4'
+    step.outlier_detection.backg = 0.0
+    step.outlier_detection.save_intermediate_results = False
+    step.outlier_detection.resample_data = False
+    step.outlier_detection.good_bits = 4
+    step.extract_1d.smoothing_length = 0
+    step.extract_1d.bkg_order = 0
+
+    step.run(asn_file)
+
+    # Compare level-2c product
+    fname = 'jw93065002001_02101_00001_nrca1_a3001_crfints.fits'
+    reffile = 'jw93065002001_02101_00001_nrca1_a3001_crfints_ref.fits'
+    extn_list = ['primary', 'sci', 'dq', 'err']
+
+    refname = os.path.join(_bigdata, subdir, reffile)
+
+    result = perform_FITS_comparison(fname, refname, extn_list=extn_list)
+    assert result.identical, result.report()
+
+    # compare the output files - use this exact command
+    # DISABLED
+    # filecomp.compare_files(output, (__file__, testname), tda=tda,)
+
+
+def test_tso3_pipeline2(_bigdata):
+    """Regression test of calwebb_tso3 pipeline on NIRCam simulated data.
+
+    Scaled imaging mode outlier_detection will be tested here.
+    """
+    testname = "test_tso3_pipeline2"
+
+    # Define where this test data resides in testing tree
+    subdir = 'pipelines/nircam_caltso3/'  # Can NOT start with path separator
+
+    # You need to have a tda dict for:
+    #  - recording information to make FlagOK work
+    #  - recording parameters to the task as attributes
+    global tda
+    tda = {}
+
+    output = [
+        # one dict for each output file to compare (i.e. each <val>)
+        {
+            'file'      : 'jw93065-a3002_t1_nircam_f150w-wlp8_phot.ecsv',
+            'reference' : os.path.join(_bigdata,subdir,'jw93065-a3002_t1_nircam_f150w-wlp8_phot_ref.ecsv'),
+            'comparator': 'diff',
+            'args'      : {},
+        }
+    ]
+
+    try:
+        os.remove("jw93065002002_02101_00001_nrca1_a3002_crfints.fits")
+        os.remove("jw93065-a3002_t1_nircam_f150w-wlp8_phot.ecsv")
+    except Exception:
+        pass
+
+    # Run pipeline step...
+    asn_file = os.path.join(_bigdata, subdir,
+                            "jw93065-a3002_20170511t111213_tso3_001_asn.json")
+    Tso3Pipeline.call(asn_file)
+
+    # Compare level-2c product
+    fname = 'jw93065002002_02101_00001_nrca1_a3002_crfints.fits'
+    reffile = 'jw93065002002_02101_00001_nrca1_a3002_crfints_ref.fits'
+    extn_list = ['primary', 'sci', 'dq', 'err']
+
+    refname = os.path.join(_bigdata, subdir, reffile)
+
+    result = perform_FITS_comparison(fname, refname, extn_list=extn_list)
+    assert result.identical, result.report()
+
+    # compare the output files - use this exact command
+    # DISABLED
+    # filecomp.compare_files(output, (__file__, testname), tda=tda,)
+
+
+# Utility function to simplify FITS comparisons
+def perform_FITS_comparison(filename, refname, **pars):
```

Comment:
Ah, looks like you already have that `fitsdiff` helper I mentioned earlier in this review. So, I guess, the work now is simply to get all the other tests to use this, as applicable.